app-backend/http4s-util/src/main/scala/com/rasterfoundry/http4s/TracedHTTPRoutes.scala

```diff
@@ -32,34 +32,49 @@ object TracedHTTPRoutes {
   ): Kleisli[OptionT[F, ?], AuthedRequest[F, User], Response[F]] = {
     Kleisli[OptionT[F, ?], AuthedRequest[F, User], Response[F]] { authedReq =>
       val req = authedReq.req
-      val operationName = "http4s-incoming-request"
-      val traceHeader =
+      val operationName = "http4s-request"
+      val tags = Map(
+        "http_method" -> req.method.name,
+        "request_url" -> req.uri.path.toString,
+        "environment" -> Config.environment
+      ).combine {
         req.headers.get(CaseInsensitiveString("X-Amzn-Trace-Id")) match {
-          case Some(header) => header.value
-          case _            => ""
+          case Some(header) => Map("amazon_trace_id" -> header.value)
+          case _            => Map.empty[String, String]
         }
-      val tags = Map(
-        HTTP_METHOD.getKey -> req.method.name,
-        HTTP_URL.getKey -> req.uri.path.toString,
-        "X-Amzn-Trace-Id" -> traceHeader
-      )
+      }
+
+      def transformResponse(context: TracingContext[F]): F[Option[Response[F]]] = {
+        val tracedRequest = AuthedTraceRequest[F](authedReq, context)
+        val responseOptionWithTags = routes.run(tracedRequest) semiflatMap {
+          response =>
+            val traceTags = Map(
+              "http_status" -> response.status.code.toString,
+            ) ++ tags ++
+              response.headers.toList
+                .map(h => (s"response_header_${h.name}" -> h.value))
+                .toMap
+            context
+              .addTags(traceTags)
+              .map(_ => response)
+        }
+        responseOptionWithTags.value
+      }
 
       OptionT {
-        builder(operationName, tags) use { context =>
-          val tracedRequest = AuthedTraceRequest[F](authedReq, context)
-          val responseOptionWithTags = routes.run(tracedRequest) semiflatMap {
-            response =>
-              val traceTags = Map(
-                HTTP_STATUS.getKey() -> response.status.code.toString
-              ) ++ tags ++
-                response.headers.toList
-                  .map(h => (s"http.response.header.${h.name}" -> h.value))
-                  .toMap
-              context
-                .addTags(traceTags)
-                .map(_ => response)
+        builder match {
+          case b: XRayTracer.XRayTracingContextBuilder[F] => {
```

Comment:
This worked out pretty well and makes me optimistic about how well we'll be able to integrate with other tracing syncs if we need to sometime later

Reply:
Yeah, it wasn't too bad - only issue is that there is a bit of boilerplate in `XRayTracingContext` when we want to add something like `XrayHttp` and we'll have to do another match if we want to add custom methods for adding/updating values (like we would want to do for the response here)